pkg/cache/scheduler/fair_sharing.go

```diff
@@ -38,17 +39,81 @@ type dominantResourceShareNode interface {
 	hierarchicalResourceNode
 }
 
-// dominantResourceShare returns a value ranging from 0 to math.MaxInt,
+// DRS contains the DominantResourceShare for some
+// node, with convenence methods for precise comparison.
+type DRS struct {
+	fairWeight       *resource.Quantity
+	unweightedRatio  float64
+	dominantResource corev1.ResourceName
+}
+
+// NegativeDRS is used as a starting point for comparisons.
+func NegativeDRS() DRS {
```

Comment:
nit/question, maybe more natural to make it `func (a DRS) Negate() DRS`?

Reply:
Not a blocker by any means, I was just thinking it might be better interface, but Im not convinced it is worth the time.Up to you